core/controllers/acl_decorators_test.py

```diff
@@ -53,9 +53,174 @@
+            response = self.get_json('/mock')
+        self.assertTrue(response['success'])
+
+
+class IsSourceMailChimpDecoratorTests(test_utils.GenericTestBase):
+    """Tests for open access decorator."""
+
+    user_email = 'user@example.com'
+    username = 'user'
+    secret = 'webhook_secret'
+    invalid_secret = 'invalid'
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+        URL_PATH_ARGS_SCHEMAS = {
+            'secret': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+        HANDLER_ARGS_SCHEMAS = {'GET': {}}
+
+        @acl_decorators.is_source_mailchimp
+        def get(self, secret):
+            return self.render_json({'secret': secret})
+
+    def setUp(self):
+        super(IsSourceMailChimpDecoratorTests, self).setUp()
+        self.mock_testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock_secret_page/<secret>', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+
+    def test_error_when_mailchimp_webhook_secret_is_none(self):
+        testapp_swap = self.swap(self, 'testapp', self.mock_testapp)
+
+        with testapp_swap:
+            response = self.get_json(
+                '/mock_secret_page/%s' % self.secret,
+                expected_status_int=404
+            )
+
+        error_msg = (
+            'Could not find the page http://localhost'
+            '/mock_secret_page/%s.' % self.secret
+        )
+        self.assertEqual(response['error'], error_msg)
+        self.assertEqual(response['status_code'], 404)
+
+    def test_error_when_given_webhook_secret_is_invalid(self):
+        testapp_swap = self.swap(self, 'testapp', self.mock_testapp)
+        mailchimp_swap = self.swap_to_always_return(
+            feconf, 'MAILCHIMP_WEBHOOK_SECRET', value=self.secret)
+
+        with testapp_swap, mailchimp_swap:
+            response = self.get_json(
+                '/mock_secret_page/%s' % self.invalid_secret,
+                expected_status_int=404
+            )
+
+        error_msg = (
+            'Could not find the page http://localhost'
+            '/mock_secret_page/%s.' % self.invalid_secret
+        )
+        self.assertEqual(response['error'], error_msg)
+        self.assertEqual(response['status_code'], 404)
+
+    def test_no_error_when_given_webhook_secret_is_valid(self):
+        testapp_swap = self.swap(self, 'testapp', self.mock_testapp)
+        mailchimp_swap = self.swap(
+            feconf, 'MAILCHIMP_WEBHOOK_SECRET', self.secret)
+
+        with testapp_swap, mailchimp_swap:
+            response = self.get_json(
+                '/mock_secret_page/%s' % self.secret,
+                expected_status_int=200
+            )
+
+        self.assertEqual(response['secret'], self.secret)
+
+
+class ViewSkillsDecoratorTests(test_utils.GenericTestBase):
     """Tests for play exploration decorator."""
```

Comment:
Docstring seems wrong. 

Please check your docstrings and test names. They are important for developers to understand your work.

Reply:
Sorry, done by mistake.
Fixed this and cross-checked all the other class names, docstrings and test names.